sdk/keyvault/Azure.Security.KeyVault.Keys/src/DeletedKey.cs

```diff
@@ -3,27 +3,66 @@
 // license information.
 
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Text.Json;
 
 namespace Azure.Security.KeyVault.Keys
 {
     public class DeletedKey : Key
     {
         public string RecoveryId { get; private set; }
 
-        public DateTime? DeletedDate { get; private set; }
+        public DateTimeOffset? DeletedDate { get; private set; }
 
-        public DateTime? ScheduledPurgeDate { get; private set; }
+        public DateTimeOffset? ScheduledPurgeDate { get; private set; }
 
-        public DeletedKey(string name, string recoveryId, DateTime? deletedDate, DateTime? scheduledPurge)
+        public DeletedKey(string name) : base(name) { }
+
+        public DeletedKey (string name, string recoveryId, DateTimeOffset? deletedDate, DateTimeOffset? scheduledPurge)
             : base(name)
         {
             RecoveryId = recoveryId;
             DeletedDate = deletedDate;
             ScheduledPurgeDate = scheduledPurge;
         }
+
+        internal override void WriteProperties(ref Utf8JsonWriter json)
+        {
+            base.WriteProperties(ref json);
+
+            if (RecoveryId != null)
+            {
+                json.WriteString("recoveryId", RecoveryId);
```

Comment:
For perf, the property name string literals should be stored as `JsonEncodedText` statics to avoid escaping checks.

Look at the following as examples:
https://github.com/aspnet/AspNetCore/blob/3bb092bdab95997134e95d02c0eeb921b02caaca/src/SignalR/common/Http.Connections.Common/src/NegotiateProtocol.cs#L17